src/parser.rs

```diff
@@ -145,13 +145,15 @@ fn parse_msg_args(args: &[u8]) -> io::Result<ControlOp> {
         4 => (args[0], 3, Some(args[2].to_owned())),
         _ => return Err(parse_error()),
     };
-    let sid = match usize::from_str(args[1]) {
-        Ok(sid) => sid,
-        _ => return Err(parse_error()),
+    let sid = if let Ok(sid) = usize::from_str(args[1]) {
+        sid
+    } else {
+        return Err(parse_error());
```

Comment:
I assume idiomatic, but is an extra line and does not read as well IMO, just curious.

Reply:
In general I associate wildcard matches with bugs over time. I agree that the match statement is nicer to look at.